File: solutions/Java/single_number_ii_bit_count.java
public class SingleNumberIIBitCount {
    public static int singleNumberIIBitCount(int[] nums) {
        int[] bitCounts = new int[32];
        for (int num : nums) {
            for (int i = 0; i < 32; i++) {
                if ((num & (1 << i)) != 0) {
                    bitCounts[i]++;
                }
            }
        }

        int result = 0;
        for (int i = 0; i < 32; i++) {
            if (bitCounts[i] % 3 != 0) {
                result |= (1 << i);
            }
        }

        return result;
    }

    public static void main(String[] args) {
        int[] nums = {2, 2, 3, 2};
        System.out.println(singleNumberIIBitCount(nums));  // 3
    }
}

File: solutions/Rust/single_number_ii_bit_count.rs
fn single_number_ii_bit_count(nums: Vec<i32>) -> i32 {
    let mut bit_counts = vec![0; 32];
    for num in nums {
        for i in 0..32 {
            if (num & (1 << i)) != 0 {
                bit_counts[i] += 1;
            }
        }
    }

    let mut result = 0;
    for i in 0..32 {
        if bit_counts[i] % 3 != 0 {
            result |= 1 << i;
        }
    }

    if result >= (1 << 31) {
        result -= 1 << 32;
    }
    result
}

fn main() {
    println!("{}", single_number_ii_bit_count(vec![2, 2, 3, 2]));  // 3
}
